code/controllers/subsystem/timer.dm

```diff
@@ -246,7 +246,7 @@ var/datum/subsystem/timer/SStimer
 	if (callBack && callBack.object && callBack.object != GLOBAL_PROC && callBack.object.active_timers)
 		callBack.object.active_timers -= src
 		UNSETEMPTY(callBack.object.active_timers)
-
+		qdel(callBack)
```

Comment:
CallBack intentionally doesn't use qdel. if there is a reference to it, it's still needed.